### ISSUE TITLE
Enable codegen in CI

### ISSUE THREAD

**Issue by RolandKluge** [eMoflon/emoflon-core#32]
@anthonyanjorin FYI

Adding generated code to the repo is ugly.
Therefore, we should trigger codegen on the CI server.
The following steps will be necessary:
1. Provide a custom Docker image with a stable eMoflon Core version installed
  * The Eclipse should contain https://github.com/seeq12/eclipse-import-projects-plugin
1. Create an Eclipse workspace that contains all plugins in emoflon-core
   * Maybe we can even automate this step?
1. Add a headless build to the CI workflow
   * See https://help.eclipse.org/oxygen/index.jsp?topic=%2Forg.eclipse.jdt.doc.isv%2Fguide%2Fjdt_apt_building_with_apt.htm

 ```
workspacePath="../.."
repositoryRoot="."

# Create workspace
$ECLIPSE_HOME/eclipse -nosplash -application org.eclipse.jdt.apt.core.aptBuild -data $workspacePath

# Import all projects
$ECLIPSE_HOME/eclipsec -nosplash -application com.seeq.eclipse.importprojects.headlessimport -data $workspacePath -import $repositoryRoot
# Repeat if necessary: 
# $eclipseHome/eclipsec -nosplash -application com.seeq.eclipse.importprojects.headlessimport -data $workspacePath -import $repositoryRoot

# Invoke build
$ECLIPSE_HOME/eclipse -nosplash -application org.eclipse.jdt.apt.core.aptBuild -data $workspacePath
```

**Comment by anthonyanjorin** [comment]
I have to admit all this stuff is a bit scary...  Although CI is very cool, the cost/benefit here for emoflon-core is not (yet) very convincing for me.  Simply opening a new workspace and importing a single PSF file works just fine.

**Comment by RolandKluge** [comment]
I think that using CI ensures that we have reproducible builds and automated test runs.
The CI of emoflon-core has saved me at least once to commit code that does not pass the unit tests.
Let's see how far I will get; the first experiments look promising.

**Comment by RolandKluge** [comment]
Solved in PR eMoflon/emoflon-core#35
